include/seqan3/alignment/pairwise_alignment_edit_distance_unbanded.hpp

```diff
@@ -0,0 +1,551 @@
+    struct state_type
+    {
+        //!\copydoc pairwise_alignment_edit_distance_unbanded::vp
+        std::vector<word_type> vp{};
+        //!\copydoc pairwise_alignment_edit_distance_unbanded::vn
+        std::vector<word_type> vn{};
+    };
+    //!\brief The collection of each computation step.
+    std::vector<state_type> states{};
+    //!\brief Add a computation step
+    void add_state()
+    {
+        states.push_back(state_type{vp, vn});
+    }
+
+public:
+
+    /*!\name Constructors
+     * \{
+     */
+    /*!\brief Constructor
+     * \param[in] _database \copydoc database
+     * \param[in] _query    \copydoc query
+     * \param[in] _config   \copydoc config
+     */
+    pairwise_alignment_edit_distance_unbanded(
+        database_t _database,
+        query_t _query,
+        align_config_t _config)
+        : database{std::forward<database_t>(_database)},
+          query{std::forward<query_t>(_query)},
+          config{std::forward<align_config_t>(_config)},
+          _score{query.size()},
+          database_it{std::cbegin(database)},
+          database_it_end{std::cend(database)}
+    {
+        static constexpr std::size_t alphabet_size = alphabet_size_v<query_alphabet_type>;
+
+        if constexpr(use_max_errors)
+            max_errors = config.max_errors;
+
+        unsigned block_count = (query.size() - 1 + word_size) / word_size;
+        score_mask = (word_type)1 << ((query.size() - 1 + word_size) % word_size);
+        last_score_mask = score_mask;
+        last_block = block_count - 1;
+
+        if constexpr(use_max_errors)
+        {
+            // localMaxErrors either stores the maximal number of _score (me.max_errors) or the needle size minus one.
+            // It is used for the mask computation and setting the initial score (the minus one is there because of the Ukkonen trick).
+            unsigned localMaxErrors = std::min<unsigned>(max_errors, query.size() - 1);
+            score_mask = (word_type)1 << (localMaxErrors % word_size);
+            last_block = std::min(localMaxErrors / word_size, block_count - 1);
+            _score = localMaxErrors + 1;
+        }
+
+        word_type vp0{~static_cast<word_type>(0)};
+        word_type vn0{0};
+
+        vp.resize(block_count, vp0);
+        vn.resize(block_count, vn0);
+        bit_masks.resize((alphabet_size + 1) * block_count, 0);
+
+        // encoding the letters as bit-vectors
+        for (unsigned j = 0; j < query.size(); j++)
+        {
+            unsigned i = block_count * to_rank(query[j]) + j / word_size;
+            bit_masks[i] |= (word_type)1 << (j % word_size);
+        }
+
+        add_state();
+    }
+    //!\}
+
+private:
+
+    /*!\todo describe me
+     */
+    template <bool with_overflow_check>
+    inline void compute_step(word_type b, word_type & hp, word_type & hn, word_type & vp, word_type & vn, word_type & carry_d0, word_type & carry_hp, word_type & carry_hn)
+    {
+        word_type x, d0, t;
```

Comment:
:dizzy_face: I hope you will change the names of those variables for the final version? Otherwise I at least won't review this code..

Reply:
they are directly taken from the paper every rename would be confusing to whoever has read the paper.